examples/guessing_game.porth

```diff
@@ -0,0 +1,78 @@
+include "std.porth"
+
+//mem 0 = rand
+//mem 1 = guess (64bit)
+//mem 10 = digit_num (64bit)
+//mem 20-255 = s
+
+// drop argc
+drop
+
+// "random" value (from pointer)
+dup 24 shr 127 band 
+mem swap .
+
+"Guess a number from 0 to 255\n" stdin write
+
+1 while dup do
+    mem 1 + 0 .64
+    mem 10 + 1 .64
+
+    //read stdin
+    255 mem 20 + stdout SYS_read syscall3
```

Comment:
Why are you reading from `stdout`? you should read from `stdin` and write to `stdout` right?

Reply:
Yes, thanks. Didn't notice it, because it works anyway lol.
Will fix that

Reply:
No problem, this works because you're not redirecting the the stdout/stderr/stdin. Try something like this `./examples/guessing_game 2>&1 | tee -a some_file` and you program will segfault.

Reply:
Makes sense. I also wrote stdin in place of stdout in some other places :| 
All fixed now